config.c

```diff
@@ -1268,16 +1277,14 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
 						newlog->compress_prog);
 
 			 		compresscmd_base=strdup(basename(newlog->compress_prog));
-			 		i=0; /* have to check whether we may do this! */
 			 		/* we check whether we changed the compress_cmd. In case we use the apropriate extension
 			 		   as listed in compress_cmd_list */
-			 		while ((i>=0)&&(strcmp(compress_cmd_list[i][0], "EOLIST"))){
-			 		    if (0==strcmp(compress_cmd_list[i][0], compresscmd_base)){
-			 			newlog->compress_ext=strdup((char *)compress_cmd_list[i][1]);
-			 			message(MESS_DEBUG, "compress_ext was changed to %s\n", newlog->compress_ext);
-			 			i=-10; /* terminate loop! */
-			 		    }
-			 		    i++;
+					for(i=0; compress_cmd_list[i][0] == NULL; i++) {
```

Comment:
The loop condition seems to be negated by mistake.  How are you confirming that the resulting code works as expected after your recent changes?